internal/lightcone/destruction/theunreachableside/theunreachableside.go

```diff
@@ -0,0 +1,84 @@
+package theunreachableside
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	side    key.Modifier = "the-unreachable-side"
+	dmgBuff key.Modifier = "the-unreachable-side-dmg-buff"
+)
+
+// Increases the wearer's CRIT rate by 18% and increases their Max HP by 18%.
+// When the wearer is attacked or consumes their own HP, their DMG increases by 24%.
+// This effect is removed after the wearer uses an attack.
+
+func init() {
+	lightcone.Register(key.TheUnreachableSide, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        5,
+		Path:          model.Path_DESTRUCTION,
+		Promotions:    promotions,
+	})
+	modifier.Register(side, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAfterBeingAttacked: buffOnAttacked,
+			OnHPChange:           buffOnHPConsume,
+			OnAfterAttack:        removeBuff,
+		},
+	})
+	modifier.Register(dmgBuff, modifier.Config{
+		Stacking: modifier.ReplaceBySource,
+	})
+}
+
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	crNHpAmt := 0.15 + 0.03*float64(lc.Imposition)
+	dmgAmt := 0.20 + 0.04*float64(lc.Imposition)
+	engine.AddModifier(owner, info.Modifier{
+		Name:   side,
+		Source: owner,
+		Stats: info.PropMap{
+			prop.CritChance: crNHpAmt,
+			prop.HPPercent:  crNHpAmt,
+		},
+		State: dmgAmt,
+	})
+}
+
+func buffOnAttacked(mod *modifier.Instance, e event.AttackEnd) {
+	dmgAmt := mod.State().(float64)
+	// add damage buff here.
+	addDmgBuff(mod.Engine(), dmgBuff, mod.Owner(), dmgAmt)
+}
+
+func buffOnHPConsume(mod *modifier.Instance, e event.HPChange) {
+	// only add dmg buff if hp change is not from being attacked.
+	// NOTE : confirm if need to check if only apply on hp decrease(not on heal) or all hpChange
+	// first glance at DM don't seem to check for it.
+	if e.IsHPChangeByDamage {
+		return
+	}
+	dmgAmt := mod.State().(float64)
+	// add damage buff here.
+	addDmgBuff(mod.Engine(), dmgBuff, mod.Owner(), dmgAmt)
+}
+
+func removeBuff(mod *modifier.Instance, e event.AttackEnd) {
+	mod.Engine().RemoveModifierFromSource(mod.Owner(), mod.Source(), dmgBuff)
+}
```

Comment:
nit: cleaner to have this on the `dmgBuff` modifier and have it just do `mod.RemoveSelf()`